prov/efa/src/rdm/efa_rdm_ep_utils.c

```diff
@@ -231,18 +229,33 @@ int efa_rdm_ep_post_user_recv_buf(struct efa_rdm_ep *ep, struct efa_rdm_ope *rxe
 	pkt_entry->alloc_type = EFA_RDM_PKE_FROM_USER_BUFFER;
 	pkt_entry->flags = EFA_RDM_PKE_IN_USE;
 	pkt_entry->next = NULL;
+	pkt_entry->ep = ep;
 	/*
 	 * The actual receiving buffer size (pkt_size) is
-	 *    rxe->total_len - sizeof(struct efa_rdm_pke)
+	 *    (total IOV length) - sizeof(struct efa_rdm_pke)
 	 * because the first part of user buffer was used to
 	 * construct pkt_entry. The actual receiving buffer
 	 * posted to device starts from pkt_entry->wiredata.
 	 */
-	pkt_entry->pkt_size = rxe->iov[0].iov_len - sizeof(struct efa_rdm_pke);
-
+	pkt_entry->pkt_size = ofi_total_iov_len(rxe->iov, rxe->iov_count) - sizeof *pkt_entry;
 	pkt_entry->ope = rxe;
 	rxe->state = EFA_RDM_RXE_MATCHED;
 
+	if (ep->user_info->mode & FI_MSG_PREFIX) {
```

Comment:
For prefix mode, we only need to walk the iov. The `pkt_entry->payload` (and below) assignments should be general for non-prefix mode as well

Reply:
Maybe, but you at least avoid calling `ofi_total_iov_len()` (linear time) for the non-prefix case